tokio/src/runtime/basic_scheduler.rs

```diff
@@ -1,22 +1,36 @@
-use crate::park::{Park, Unpark};
+use crate::loom::sync::Mutex;
+use crate::park::{CachedParkThread, Park, Unpark};
 use crate::runtime;
 use crate::runtime::task::{self, JoinHandle, Schedule, Task};
+use crate::sync::Notify;
 use crate::util::linked_list::{Link, LinkedList};
-use crate::util::{waker_ref, Wake};
+use crate::util::{waker_ref, Wake, WakerRef};
 
-use std::cell::RefCell;
 use std::collections::VecDeque;
 use std::fmt;
 use std::future::Future;
-use std::sync::{Arc, Mutex};
+use std::sync::Arc;
 use std::task::Poll::Ready;
 use std::time::Duration;
+use std::{cell::RefCell, sync::PoisonError};
 
 /// Executes tasks on the current thread
-pub(crate) struct BasicScheduler<P>
-where
-    P: Park,
-{
+pub(crate) struct BasicScheduler<P: Park> {
+    /// Inner state guarded by a mutex that is shared
+    /// between all `block_on` calls.
+    inner: Mutex<Inner<P>>,
+
+    /// Sendable task spawner
+    spawner: Spawner,
+}
+
+struct Inner<P: Park> {
+    scheduler: Option<Scheduler<P>>,
+    notify: Arc<Notify>,
```

Comment:
Can the `Notify` instance be stored on `BasicScheduler` outside of the `Mutex` w/o an additional `Arc`? I'm not seeing a reason for it.

If you can do it, you shouldn't need this struct anymore and can do: `Scheduler` -> `Inner`

Reply:
Yes....totally forgot we are not wrapping things in Arc, good catch 👍